### PR TITLE
Rollaxis

### PR DESCRIPTION
**Pull request by unnonouno**
I implemented rollaxis function.

### CODE REVIEW

**Comment by rezoo** [comment]
LGTM 👍

File: chainer/functions/array/rollaxis.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+from chainer import function
+from chainer.utils import type_check
+
+
+class Rollaxis(function.Function):
+
+    """Roll axis of an array."""
+
+    def __init__(self, axis, start):
+        self.axis = int(axis)
```
Comment (delta2323) [inline chainer/functions/array/rollaxis.py]: Is there any reason why you apply `int` function to `axis`and `start`? Document imposes `axis` and `start` be integers. So we need not to convert them.

**Comment by delta2323** [comment]
Please check the case when `start` and `axis` are invalid.

File: tests/chainer_tests/functions_tests/array_tests/test_rollaxis.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+from chainer.testing import attr
+
+
+@testing.parameterize(
+    {'axis': 0, 'start': 2, 'out_shape': (3, 2, 4)},
+    {'axis': 2, 'start': 0, 'out_shape': (4, 2, 3)},
+    {'axis': 1, 'start': 1, 'out_shape': (2, 3, 4)},
+    {'axis': -3, 'start': 2, 'out_shape': (3, 2, 4)},
+    {'axis': -1, 'start': 0, 'out_shape': (4, 2, 3)},
+    {'axis': -2, 'start': -2, 'out_shape': (2, 3, 4)},
```
Comment (delta2323) [inline tests/chainer_tests/functions_tests/array_tests/test_rollaxis.py]: It is desirable to check all patterns to subtract or not subtract `ndim` from `axis` and/or `start`.

Specifically we should parameterize like this (sorry, I cannot come up with appropriate parameter names):
```
testring.parameterize_dict(
    [{'axis': 0, 'start': 2}, {'axis': 2, 'start': 0}, {'axis': 1, 'start': 1}],
    [{'axis_plus': 0, 'start_plus': 0}, {'axis_plus': 0, 'start_plus': -3}, {'axis_plus': -3, 'start_plus': 0}, {'axis_plus': -3 'start_plus': -3}])
```

We can also write second line as `testing.product({'axis_plus': [0, -3], 'start_plus': [0, -3]})`. But I think it incurs readability...

**Comment by delta2323** [comment]
@unnonouno @rezoo I added comments. Could you check them?

File: tests/chainer_tests/functions_tests/array_tests/test_rollaxis.py (hunk 2 above)
Comment (unnonouno) [inline tests/chainer_tests/functions_tests/array_tests/test_rollaxis.py]: I don't think we need to test all patterns. We need to remove non-trivial logic from the test code as possible as we can.

File: chainer/functions/array/rollaxis.py (hunk 1 above)
Comment (unnonouno) [inline chainer/functions/array/rollaxis.py]: We need to check if arguments are int. But, OK, i'll replace this code with a simple if statement.

File: tests/chainer_tests/functions_tests/array_tests/test_rollaxis.py (hunk 2 above)
Comment (delta2323) [inline tests/chainer_tests/functions_tests/array_tests/test_rollaxis.py]: I agree with you in that we do not have to test trivial cases nor to test those cases which are essentially same. But I believe the test cases I listed runs the different code path, except one combination mention below.

Namely, we have combination of the following branchings 
* `start` is positive or negative
* `axis` is positive or negative
* (modified) `start` is bigger or smaller than (modified) `axis`.
Each test case reflects one of each situation listed.

 `(axis, start) = (0, 2)` and `(axis, start) = (1, 1)` runs the same code path. So we can safely remove these cases.

**Comment by delta2323** [comment]
@unnonouno LGTM. I commented on the coverage of test cases [here](https://github.com/pfnet/chainer/pull/1401#discussion_r72017385), but I did not notice you added the test cases when I wrote it. Your test seems sufficient.

**Pull request merged**